### ISSUE TITLE
🔧 Suggestions for the UI

### ISSUE THREAD

**Issue by newhinton** [automatic-ripping-machine/automatic-ripping-machine#925]
### Is this feature missing in the latest version?

- [X] I'm using the latest release

### Is your feature request related to a problem? Please describe.

I am using ARM for quite a while, and there is nothing like it. It just works!

However, the UI is sometimes a bit hard to navigate, especially the log-part. I would like to make some suggestions, but before doing so i want to know if you are interested in any suggestions.

### Describe the solution you'd like?

When you accept suggestions, i will try to mock-up some ideas i have to make navigation more easy.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by microtechno9000** [comment]
we welcome any suggestions, as users of the system we all know what works and doesn't

The devs might not get to suggestions right away, but they get tagged and incorporated slowly.

So please add away

**Comment by newhinton** [comment]
Thanks!


So first comment, i have some suggestions for the main view.

Beginning with the list of jobs:

![image](https://github.com/automatic-ripping-machine/automatic-ripping-machine/assets/25279821/c80d8d33-f863-4952-a444-6f7a31547d09)

This is organized in a grid view, but i think a table/list would be better:
![image](https://github.com/automatic-ripping-machine/automatic-ripping-machine/assets/25279821/6b361840-f6e2-4a7f-a35e-42d03820b0fc)

This does not have the problem of shifting elements or unused screen-space like the grid. Much of the information in the current card is also not that important, so it could be hidden in a modal that can be opened by a button. In general i would also advise to use more iconograpyh, because that usually makes sites look way better without much work. (I'd suggest https://forkaweso.me/Fork-Awesome/, they are rather easy to use and have a lot of icons!)

Many of the buttons can be "hidden" in the More-dropdown, ownly showing a select few important buttons in full.

Second:

The "header" above the grid:
The ARM-Logo and welcome message could be removed completely. If wanted, the branding (logo AND name) can be moved to the fixed navigation bar, but i dont think it is nessessary to have two logos on one page. This also recovers some screen-space. 

Third: 

Move the system-block somewhere else. One idea i had was to split the screen into 2 colums (75% and 25% width respectively) where the first colum is the job list, and the second column contains system information. 

Similar to this:
![image](https://github.com/automatic-ripping-machine/automatic-ripping-machine/assets/25279821/7d77ffd0-622e-4905-ba83-a9fcdfd408a8)
Also a lot of this information could be displayed more compact, eg. by removing line-seperators where not nessessary.

I would also highly suggest here to use the same icon-style everywhere.

**Comment by newhinton** [comment]
I also think the bottom-bar with the github link deserves a more prominent place:
![te2st](https://github.com/automatic-ripping-machine/automatic-ripping-machine/assets/25279821/f10b72fb-4307-4fe2-90f2-6e0220c5c33f)

(icon is from forkawesome)

**Comment by microtechno9000** [comment]
nice work 

your welcome to fork the project, make changes and raise a PR

we prefer smaller changes in PRs to minimise the review impact

**Comment by newhinton** [comment]
@microtechno9000 is there a way to get a "fake" joblist for development purposes? So that i always get running jobs?

**Comment by microtechno9000** [comment]
No, It is something I would like

there are some development tools in the devtools folder but it doesn't inject any data or states.

I started looking at flask unit testing which would allow for state setting and testing, but haven't had time to set anything up.

**Comment by newhinton** [comment]
@microtechno9000 While digging around the UI code, i have a question. Currently ARM uses a mix of server side rendering and "plain" javascript injections. This makes code maintenande rather hard.

Would it be a good project to decouple the UI and the Server, and rewriting the UI to be a fully client side application?
This would need us to define some api's (like the joblist or history) but would allow us to develop the webui independendly. It also allows other people to write other clients.

It also allows us to make heavy use of frameworks like vue, which makes maintenance a whole lot easier. However, if i try that, it would not be a small PR.

**Comment by microtechno9000** [comment]
agreed, that is currently on the cards. However as you may have noted in the code the server and UI are very intertwined. Over the past year each PR has been aiming towards that goal to make it easier to manage.

To make it easier to incorporate PRs we aim for small updates rather than a monolithic review.

**Comment by newhinton** [comment]
>  However as you may have noted in the code the server and UI are very intertwined

Yeah, that was why i was asking :D

> Have a look at https://github.com/automatic-ripping-machine/automatic-ripping-machine/issues/589 for some context.

This is the backend, i dont want to touch that ;) . The Frontend seems "rather" decoupled, what i mean by that is that it seems to me the actual webserver just reads data from the db-file, which is independently written to by the ripper.

What i want to try is to rewrite that part. However, i dont think it will be possible to not have a big PR for that, since as soon as i introduce vue, i have to fully replace the ui AND the web-api. 

On a bright note: The new API and UI should be much easier to review than the current one, since it should be way more modular.

**Comment by microtechno9000** [comment]
With the current DB issues, in that SQlite dB files only allow for a single read write, some work was being done on moving the db to MYSQL via a new container. 

1337server has started on migrating to vue, have a look at the vue branch

https://github.com/automatic-ripping-machine/automatic-ripping-machine/tree/arm-vuejs

I am keen to decouple the UI to a standalone image separated from the server code. Some initial issues I see are the flask db models and the calls between the server and UI utils.

Welcome to make a start and help out. See what you can do that can be done in a smaller PR, but agree the main bulk of code breakout would be a rather large change.

**Comment by newhinton** [comment]
> 1337server has started on migrating to vue, have a look at the vue branch

This looks great! Exactly what i wanted to achieve! I will look at how i can contribute to that branch if @1337-server wants me to help out!

**Comment by 1337-server** [comment]
That sounds great, any extra help is always appreciated!

I'm still needing to sort the issue of the frontend being hard-coded to a set IP on the backend, which obviously doesn't work with dockers/other computers. But I haven't had much free time recently.

**Comment by newhinton** [comment]
If i understand you correctly, i think this should work:

```
// Set global arm api backend link
app.config.globalProperties.armapi = 'http://' +window.location.hostname+ ':8890'

```

This allows the ui to query the api based on it's own url. I have used this trick for a different project.

**Comment by 1337-server** [comment]
> app.config.globalProperties.armapi = 'http://' +window.location.hostname+ ':8890'

This looks good, I've got some time this weekend, so ill get round to fully testing it out. Thanks.

BTW if you want, you can submit PR's to the vuejs branch

**Comment by 1337-server** [comment]
Just wanted to add that I've added your fix to the branch, it works!
Thanks for that, I've also pushed a lot of changes to the branch that I had laying uncommitted on my dev machine.

**Comment by uprightbass360** [comment]
relies on a custom version of arm but already captures many workflows : https://github.com/uprightbass360/automatic-ripping-machine-ui

**Comment by newhinton** [comment]
This looks interesting, but at the moment i don't want to touch my setup, its a bit finnicky.

@uprightbass360 Though a question: Is there a reason you don't try to get some of the stuff merged back upstream?

**Comment by uprightbass360** [comment]
Merging back means too many touch-points at the moment. To merge this back as is would at least require a completely new api backend and with that comes contracts and architectural differences. I will need more buy-in on the backend and arm core.  I definitely plan on it but I'd love to get some more eyes on it before I start bringing in everything. My first step is going to be getting involved in the v3 initiative and the discord.
